libs/ledger/src/chain/main_chain.cpp

```diff
@@ -1105,6 +1177,33 @@ bool MainChain::UpdateTips(IntBlockPtr const &block)
   assert(block->total_weight != 0);
   assert(block->block_number != 0);
 
+  if (enable_stutter_removal_)
+  {
+    // if tip is from miner who has previously produced a block for this round
+    // do not add new block to tips
+    if (stutter_blocks_.find(block->block_number) != stutter_blocks_.end() &&
+        stutter_blocks_.at(block->block_number).find(block->weight) !=
+            stutter_blocks_.at(block->block_number).end())
+    {
+      // Mark that duplicate has been seen
+      stutter_blocks_.at(block->block_number).at(block->weight) = true;
+
+      // Remove any existing tips with same height and weight
+      for (auto const &tip : tips_)
+      {
+        if (tip.second.block_number == block->block_number && tip.second.weight == block->weight)
+        {
+          auto old_heaviest = heaviest_;
+          RemoveTip(std::make_shared<Block>(*GetBlock(tip.first)));
+          return old_heaviest.hash == heaviest_.hash;
```

Comment:
It is not necessarily false as the removal of the stutter block from tips could have changed the heaviest weight tip (even if it is a lower weight to the previous stutter tip). I will add the comment and change RemoveTips to avoid the smart pointer construction.

Reply:
The return of UpdateTips is used to determine whether the updated chain should be written to file. Since moving to a lower weight tip does not necessarily correspond to a shorter chain I have left the return as it is.